test/Atmos/Parameterizations/Microphysics/KM_saturation_adjustment.jl

```diff
@@ -93,16 +93,18 @@ function nodal_update_auxiliary_state!(
 
         # saturation adjustment happens here
         ts = PhaseEquil(param_set, aux.e_int, state.ρ, aux.q_tot)
-        pp = PhasePartition(ts)
+        q = PhasePartition(ts)
 
         aux.T = ts.T
-        aux.q_vap = aux.q_tot - pp.liq - pp.ice
-        aux.q_liq = pp.liq
-        aux.q_ice = pp.ice
+        aux.q_vap = vapor_specific_humidity(q)
+        aux.q_liq = q.liq
+        aux.q_ice = q.ice
```

Comment:
```suggestion
        aux.q_vap = vapor_specific_humidity(ts)
        aux.q_liq = liquid_specific_humidity(ts)
        aux.q_ice = ice_specific_humidity(ts)
```

Reply:
On second thought, perhaps we should overload `getproperty` for `ThermodynamicState` for `tot`, `liq`, `ice` and `vap` to get what we want. I think the phases are a bit special, and I think this might make things look really nice.

Reply:
Sounds good to me. I this the readability would be better with overloading `getproperty` and keeping the ability to write `q.tot`, `q.liq` etc